### DISCUSSION TITLE
Shelly 1PM Tasmota

### DISCUSSION THREAD

**Discussion by Nazarbayevv** [arendst/Tasmota#10976] [Q&A] [answered]
Hello,

The datasheet of Shelly 1PM says, that: "Shelly 1 PM has а built-in temperature sensor for overheating protection. If the PCB temperature goes above a danger level, the device will turn off. Shelly 1 PM will turn off the circuit immediately if it detects a consumption higher than the pre-defined over-power setting"

Does temperature protection mechanism incruded to Tasmota firmware. If yes, should I turned on this or this protection actiavated in default configuration?

**Comment by arendst** [comment, marked as answer]
It's default activated

**Reply by Nazarbayevv** [reply]
Thanks. And what temperature threshold? 95 degrees?
Becouse 1PM's operating temperature scares me.

**Reply by sfromis** [reply]
Shelly has for another device (Shelly 2.5) stated that 90 degrees operation temperature is normal.

**Comment by arendst** [comment]
Default is 90 degrees but you can change it using command ``SetOption42``

**Reply by Nazarbayevv** [reply]
Thank you

**Reply by Nazarbayevv** [reply]
My house was without electricity a few about 1 hour and Energy total (consumption) reseted:(
Is it normal situation for shelly 1PM with Tasmota? Maybe there is any way to fix it?

**Reply by sfromis** [reply]
Power cuts sometimes results in fluctuations, which can make a Tasmota device count it as quick power cycle (a recovery feature). That can be controlled by SetOption65.

**Reply by Nazarbayevv** [reply]
@sfromis But I think, that in this case deveice shold restore to defaults all settings? My device just reset total power consumption, but still "Energy yesterdey", wi-fi and MQTT settings without changes.

**Reply by barbudor** [reply]
It is a know behavior, discussed already many time on Discord, that Total Energy is lost in case of power-cut.

Energy fields are continuously saved by default in the RTC RAM which survive across reboot but not in case of power loss
It is saved in Flash only if something else triggers a flash write. It is not continuously saved to flash in order to avoid wearing the flash.
EnergyToday is copied to EnergyYesterday and saved to flash at Midnight.
So if nothing happens on the device triggering the flash write (such as changing a parameter), and more over if you have enabled anti-wearing features such as `SetOption0`, this will not be saved to flash.
Better to save them on your back-end and eventually restore them upon restart with [EnergyReset](https://tasmota.github.io/docs/Commands/#energyreset)
